lib/experimental/new_embed.dart

```diff
@@ -674,12 +670,41 @@ class NewEmbed {
 // toggle that class.
 class NewEmbedTabController extends TabController {
   final MDCTabBar _tabBar;
+  final Dialog _dialog;
+  bool _userHasSeenSolution = false;
+
+  NewEmbedTabController(this._tabBar, this._dialog);
+
+  void registerTab(TabElement tab) {
+    tabs.add(tab);
 
-  NewEmbedTabController(this._tabBar);
+    try {
+      tab.onClick.listen((_) => selectTab(tab.name));
+    } catch (e, st) {
+      print('Error from registerTab: $e\n$st');
+    }
+  }
 
   /// This method will throw if the tabName is not the name of a current tab.
   @override
-  void selectTab(String tabName) {
+  Future selectTab(String tabName) async {
+    // Show a confirmation dialog if the solution tab is tapped
+    if (tabName == 'solution' && !_userHasSeenSolution) {
+      var result = await _dialog.show(
+        'Show solution?',
+        'If you just want a hint, click <span style="font-weight:bold">Cancel'
+            '</span> and then <span style="font-weight:bold">Hint</span>.',
+        yesText: "Show solution",
+        noText: "Cancel",
+      );
+      // Go back to the editor tab
+      if (result == DialogResult.no) {
```

Comment:
Could you also just do nothing here and leave the currently selected tab selected?

Reply:
no, at this point the material-components-web library has already selected the tab